src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1090,10 +1108,42 @@ - (void)_reportAttributes:(NSArray<NSDictionary<NSString *, id> *> *)attributes
     }
 }
 
+// When we receive an attribute report, check if there are any changes in parts list, server list, device type list, accepted commands list,
+//  attribute list, cluster revision or feature map attributes of the descriptor cluster. If yes, make a note that the device configuration changed.
+- (void)_noteDeviceConfigurationChanged:(NSArray<NSDictionary<NSString *, id> *> *)attributeReport
+{
+    for (NSDictionary<NSString *, id> * attribute in attributeReport) {
+        MTRAttributePath * attributePath = attribute[MTRAttributePathKey];
+
+        if (attributePath.cluster.unsignedLongValue != MTRClusterDescriptorID) {
```

Comment:
Yeah, so: this check is wrong.

Reply:
My bad.this is applicable only for parts list, device type list and server list.